Entities/Items/Sponge/Sponge.as

```diff
@@ -29,7 +29,7 @@ void onTick(CBlob@ this)
 	f32 tilesize = map.tilesize;
 
 	//absorb water
-	if (absorbed < ABSORB_COUNT)
+	if (absorbed < ABSORB_COUNT && !this.isInInventory())
```

Comment:
Would it not be better to use runflag tick_not_ininventory

Reply:
Right! I changed it now